DataDog/datadog-operator
# Rework the creds manager

### What does this PR do?

- Disable/Enable the metrics forwarder via `operatorMetricsEnabled`
- The metrics forwarder uses `pkg/config` to get creds from the operator config as a fallback
- Add a creds caching in `GetCredentials`
- Add an exp backoff in `GetCredentials`
- Disable Datadog Monitor by default
- Restart the operator if Datadog Monitor is enabled + no creds provided 

### Motivation

Follow-up on https://github.com/DataDog/datadog-operator/pull/245

### Describe your test plan

- Set the `operatorMetricsEnabled` operator argument to `false`, make sure the metrics forwarder doesn't start
- Enable Datadog Monitor by default, make sure the operator starts only when creds are provided
